### DISCUSSION TITLE
Render Elements in Edit View

### DISCUSSION THREAD

**Discussion by GraphicOwls** [notaapp/talk#284] [Feature Requests]
Hello,

I wish Nota rendered the tables in edit view. I personally never use Preview. I don't really see the point. Most things look great when editing minus a couple things. Tables as I mentioned and things like checkboxes. Tables would be much easier to work with. And I can't imagine it being too much of a hassle to remove the white background from the checkboxes. Making them look like they do in preview mode. 

Another thing I found odd was the dividers in edit mode vs preview. Headings get divider lines that are consistent with how they look in edit mode (even though I would prefer to add the dividers myself), but then divider lines that we add become extremely thick and don't match the edit mode. I'm not a fan of the inconsistency. I like how they look in edit mode. They're light, thin and clean looking.

Overall, I wish I could see the tables render in edit mode so it's much easier and quicker to work with, and the checkboxes looked more like the preview mode but in edit mode.

Thanks for reading 😉
– Noah

**Comment by marlonjames71** [comment]
Would love to see preview axed and just make the editor super pretty. It's already 80% there anyway.

If you think about it, a lot of the markdown syntax is already hidden in Nota. And a lot of those markdown elements like headings, links, bold text, italic text, tags, etc are stylized on top of that. Not having tables render feels kind of like a half baked solution maybe?

Definitely give a thumbs up for rendering everything in the editor.

**Comment by vladstudio** [comment]
A WYSIWYG table editor sounds awesome... in theory. In practice, I'm afraid, it makes Nota less of a markdown editor. And it is a very complex task all by itself.

**Comment by GraphicOwls** [comment]
Do you feel the same way about Obsidian. Obsidian has a nice plugin for tables. At least it assists with creating tables beautifully in Markdown and just making it extremely convenient and a pleasure to write. What would be great is if it renders in the editor view once your cursor leaves the table. Again making preview mode not necessary. The same way Nota works for code blocks, etc.

https://user-images.githubusercontent.com/30763670/183312161-27c32cff-81a0-4b7a-9111-4059115ec41e.mp4

**Comment by astoilkov** [comment]
We want to add tables at some point in the editor. I think this is a must.

However, tables are a really complex feature. Even displaying a table instead of the text when the cursor is not over the table is hard. This means that it will take us some time to get there.

Our roadmap also includes other improvements to the editing experience so stay tuned.

**Comment by Arbiter313** [comment]
Someone must be reading my mind because I was just going to make a post about tables. I liked the implementation in Caret and the assistance the editor gave you in creating a table and something similar in Nota would be great. I second @GraphicOwls post although I understand it could be difficult to implement. That obsidian plugin looks like it has some interesting ideas too!

**Reply by astoilkov** [reply]
We can bring the writing support we had for tables. I will soon take a look on how hard it will be.

**Comment by DutchPete** [comment]
I would really like @GraphicOwls's suggestion implemented and in that context @marlonjames71's suggestion makes a lot of sense.
